src/main/java/net/mcreator/element/types/Projectile.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.types;
+
+import net.mcreator.element.BaseType;
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.parts.MItemBlock;
+import net.mcreator.element.parts.Sound;
+import net.mcreator.element.parts.procedure.Procedure;
+import net.mcreator.element.types.interfaces.ICommonType;
+import net.mcreator.element.types.interfaces.IEntityWithModel;
+import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.resources.Model;
+
+import java.util.Collection;
+import java.util.Collections;
+
+public class Projectile extends GeneratableElement implements IEntityWithModel, ICommonType {

```

Comment:
This ME could have preview picture that is either texture of projectileItem or if customModel is selected, customModelTexture